Source/NSGridView.m

```diff
@@ -0,0 +1,1280 @@
+  if (self != nil)
+    {
+      NSDebugLog(@"%@ %@",NSStringFromClass([self class]), NSStringFromSelector(_cmd));
+      
+      if ([coder allowsKeyedCoding])
+        {
+          if ([coder containsValueForKey: @"NSGrid_content"])
+            {
+              [self setContentView: [coder decodeObjectForKey: @"NSGrid_content"]];
+              NSDebugLog(@"contentView = %@", [self contentView]);
+            }
+          if ([coder containsValueForKey: @"NSGrid_mergeHead"])
+            {
+              _mergeHead = [coder decodeObjectForKey: @"NSGrid_mergeHead"];
+            }
+          if ([coder containsValueForKey: @"NSGrid_owningRow"])
+            {
+              _owningRow = [coder decodeObjectForKey: @"NSGrid_owningRow"]; // weak
+              NSDebugLog(@"_owningRow = %@", _owningRow);
+            }
+          if ([coder containsValueForKey: @"NSGrid_owningColumn"])
+            {
+              _owningColumn = [coder decodeObjectForKey: @"NSGrid_owningColumn"]; // weak
+              NSDebugLog(@"_owningColumn = %@", _owningColumn);
+            }
+          if ([coder containsValueForKey: @"NSGrid_xPlacement"])
+            {
+              _xPlacement = [coder decodeIntegerForKey: @"NSGrid_xPlacement"];
+            }      
+          if ([coder containsValueForKey: @"NSGrid_yPlacement"])
+            {
+              _yPlacement = [coder decodeIntegerForKey: @"NSGrid_yPlacement"];
+            }  
+          if ([coder containsValueForKey: @"NSGrid_alignment"])
+            {
+              _rowAlignment = [coder decodeIntegerForKey: @"NSGrid_alignment"];
+            }      
+        }
+      else
+        {
+          [self setContentView: [coder decodeObject]];
+          ASSIGN(_mergeHead, [coder decodeObject]);
+          _owningRow = [coder decodeObject]; // weak
+          _owningColumn = [coder decodeObject]; // weak
+          [coder decodeValueOfObjCType: @encode(NSInteger)
+                                    at: &_xPlacement];
+          [coder decodeValueOfObjCType:@encode(NSInteger)
+                                    at: &_yPlacement];
+          [coder decodeValueOfObjCType:@encode(NSInteger)
+                                    at: &_rowAlignment];
+        }
+    }
+  return self;
+}
+
+- (void) dealloc
+{
+  RELEASE(_contentView);
+  RELEASE(_mergeHead);
+  [super dealloc];
+}
+
+@end
+
+/// Column ///
+@implementation NSGridColumn
+
++ (void) initialize
+{
+  if (self == [NSGridColumn class])
+    {
+      [self setVersion: 1];
+    }
+}
+
+- (void) setGridView: (NSGridView *)gv
+{
+  _gridView = gv;
+}
+
+- (NSGridView *) gridView
+{
+  return _gridView;
+}
+
+- (NSInteger) numberOfCells
+{
+  NSUInteger ci = [_gridView indexOfColumn: self];
+  NSArray *arr = [_gridView _cellsForColumnAtIndex: ci];
+  return [arr count];
+}
+
+- (NSGridCell *) cellAtIndex:(NSInteger)index
+{
+  NSUInteger ci = [_gridView indexOfColumn: self];
+  NSArray *arr = [_gridView _cellsForColumnAtIndex: ci];
```

Comment:
You do not need this intermediate array, you could compute the index of the cell directly.

Reply:
Modified to use -[NSGridView cellAtColumnIndex:rowIndex: I have left a comment in there which explains why it is done this way.